src/components/ActivityFeed/ActivityFeed.js

```diff
@@ -223,16 +227,21 @@ class ActivityFeed extends React.Component<Props, State> {
         directionSymbol = '';
       }
 
+      const fullIconUrl = iconUrl ? `${SDK_PROVIDER}/${iconUrl}?size=3` : '';
+
       const contact = contacts
         .find(({ ethAddress }) => address.toUpperCase() === ethAddress.toUpperCase()) || {};
+      const inContacts = Object.keys(contact).length !== 0;
+      const itemImage = contact.profileImage || fullIconUrl;
 
       return (
         <ListItemWithImage
           onPress={() => this.selectEvent({ ...notification, value, contact }, type, notification.status)}
           label={nameOrAddress}
-          avatarUrl={contact.profileImage}
-          navigateToProfile={Object.keys(contact).length !== 0 ? navigateToContact : null}
-          iconName={Object.keys(contact).length === 0 || showArrowsOnly ? directionIcon : null}
+          avatarUrl={itemImage}
+          navigateToProfile={inContacts ? navigateToContact : null}
+          iconName={(showArrowsOnly || !itemImage) ? directionIcon : null}
+          imageAddonIconName={(itemImage && !showArrowsOnly) ? directionIcon : undefined}
```

Comment:
Shouldn't it be `null` and not `undefined`?

Reply:
With `null` it triggers a warning for incompatible types. I just copied how we do it on other places. We could update the `ListItemWithImage` component to allow `imageAddonIconName=null`, but  I think it's not on the scope of this fix. What do you think?

Reply:
No, it's fine with undefined :)